src/quail/sensors/ADCTask.hpp

```diff
@@ -25,29 +23,31 @@ class ADCTask : public Task<2000>
     } typedef adcdata_t;
 
     /**
-     * @brief Initializes the ADC, after execution ready to process data
-     */ 
-    void initADC();
-
-    /**
-     * @brief set flag to indicate that all sensor channels have been configured
+     * @brief set flag to indicate that all sensor channels have been configured. This is meant to be called from another task (SensorTask).
      */
     void sensorsConfigured();
 
+private:
+
+    enum events{
+        NewData = 0b10,
+        SensorsReady = 0b01
+    };
```

Comment:
I like this better. Let's do something similar for all tasks with multiple active bits in their eventgroups.